### DISCUSSION TITLE
Making app url and app name dynamic?

### DISCUSSION THREAD

**Discussion by surendrakandel** [pocketbase/pocketbase#1415] [Q&A]
Can we make App url and App name dynamic? maybe this feature already exits as I saw setting's model Meta field is public. 

use-cases: 
for example, I host multiple frontend but wants to host only one instance for pocketbase, in such case I want to be able to change the app name according to my allowed frontend url (app name and app url) to catch the auth token confirmation as well as other possible dynamic routes.

**Comment by ganigeorgiev** [comment]
I'm not sure that I understand the described use-case. Could you elaborate a little more?

You can change the App name and url from the "Admin UI > Settings > application" or via the Settings APIs - https://pocketbase.io/docs/api-settings/, if that's what you are asking.

**Reply by surendrakandel** [reply]
I see that, but I want to make it possible from client side, or based on event hook and changing of values should be thread safe.

**Reply by ganigeorgiev** [reply]
I'm sorry, I don't understand. Could you elaborate a little more what you are trying to do and/or provide a code sample/pseudo-code?

**Reply by surendrakandel** [reply]
lets say I am hosting pocket base on auther.io but i am relaying on it for authentication of ecom1.com, and ecom2.com when I am sending any emails, notification or any sort of client/server driven activities i want all the origin urls, app name to reflect the ecom1.com or ecom2.com not auther.io . In this scenario, auther.io should be able to change its app name, app url, and all other relevant configs name based on if request is coming from ecom1.com or ecom2.com.

**Reply by ganigeorgiev** [reply]
Hm, in this case you'll have to use the [`OnMailer*` record event hooks](https://pocketbase.io/docs/event-hooks/#mailer-hooks) and either adjust the email that will be send or generate a completely different one:

```go
// adjust the email message that is being sent
app.OnMailerBeforeRecordVerificationSend().Add(func(e *core.MailerRecordEvent) error {
	e.Message.Subject = "Something else..."
	e.Message.HTML = strings.ReplaceAll(e.Message.HTML, "something", "else")

	return nil
})

// OR send a completely different email template
app.OnMailerBeforeRecordVerificationSend().Add(func(e *core.MailerRecordEvent) error {
	err := e.MailClient.Send(&mailer.Message{
		From:    mail.Address{Address: "from@example.com"},
		To:      mail.Address{Address: "to@example.com"},
		Subject: "YOUR_SUBJECT...",
		HTML:    "YOUR_HTML_BODY...",
	})
	if err != nil {
		return err
	}

	return hook.StopPropagation
})
```

But there is no integrated multi-app tenancy if that's what you are looking for.
